src/ft60x.rs

```diff
@@ -97,37 +105,131 @@ impl FT60x {
 
         let mut async_group = AsyncGroup::new(&self.context);
         for (i, chunk) in mut_chunks.enumerate() {
+            // The FT60x doesn't seem to like too many outstanding requests
+            if i > 500 {
+                let mut transfer = async_group.wait_any()?;
+                ensure!(
+                    transfer.buffer().len() == transfer.actual().len(),
+                    "FT60x did not return enough data. requested {} got {}",
+                    transfer.buffer().len(),
+                    transfer.actual().len()
+                );
+                collected += 1;
+            }
+
             async_group.submit(Transfer::bulk(
                 &self.device,
                 0x82,
                 chunk,
                 Duration::new(1, 0),
             ))?;
-
-            // The FT60x doesn't seem to like too many outstanding requests
-            if i > 50 {
-                let mut transfer = async_group.wait_any()?;
-                assert_eq!(transfer.buffer().len(), transfer.actual().len());
-                collected += 1;
-            }
         }
         while let Ok(mut transfer) = async_group.wait_any() {
-            assert_eq!(transfer.buffer().len(), transfer.actual().len());
+            ensure!(
+                transfer.buffer().len() == transfer.actual().len(),
+                "FT60x did not return enough data. requested {} got {}",
+                transfer.buffer().len(),
+                transfer.actual().len()
+            );
             collected += 1;
         }
-        assert_eq!(collected, mut_chunks_len);
-
+        ensure!(
+            collected == mut_chunks_len,
+            "FT60x did not answer all chunks within timeout. Requested {} got an answer for {}",
+            mut_chunks_len,
+            collected
+        );
         Ok(())
     }
 
-    /// it es recommended to request multiples of 32Kb
-    pub fn data_stream(mut self, bufsize: usize) -> Result<RingBufConsumer<Vec<u8>>> {
+    // starts a thread with which you can send empty buffers and receive full buffers from
```

Comment:
Maybe it should assert if a buffer is not a exact multiple of blocksize?

Otherwise it probably segfaults or so in `rusb`